docs/api/authentication.rst

```diff
@@ -0,0 +1,136 @@
+.. _authenticating:
+
+Authenticating to the Kinto API
+###############################
+
+First of all **Kinto doesn't handle users managements**.
+
+You do not have such thing as user creation, user removal, user
+password modifications, etc.
+
+However Kinto handle users permissions, which means users are uniquely
+identified in Kinto.
+
+
+How is that possible?
+---------------------
+
+This is possible by plugging in Kinto with an Identity provider.
+
+Multiple identity providers solutions are available such as OAuth,
+SAML, x509, Hawk sessions, or Basic Tokens.
+
+With regards to the application you are building you may want to plug
+Github, Facebook, Google, or your company identity provider.
+
+You may also want to use arbitrary tokens by making sure:
+
+ - each user has a different one;
+ - a user always uses the same token.
+
+This is what we are doing within the the scope of this documentation.
+
+We use the Basic Auth protocol like that: ``token:my-secret-token``
+
+
+How can I generate a strong unique tokens?
+------------------------------------------
+
+I would recommand you to use more than 16 random bytes digested as
+either a Base64 or a Hexadecimal string:
+
+Here are two examples using Node and Python to generate 30 random bytes tokens:
+
+.. code-block:: js
+
+    var crypto = require("crypto");
+
+	crypto.randomBytes(30).toString("hex"));
+    // 8d11dfa2ab1fab42c841bcda834a66553624b03e8ef6bb5e8ec6fdd791a9
+
+    crypto.randomBytes(30).toString("base64");
+    // nkz6+hN9KVtXX6bZ9+RPof0NgF9hmm5gFhKpWbiM
+
+.. code-block:: python
+
+    import base64, os
+
+    print(base64.b64encode(os.urandom(30)))
+    # BxFNjp97FprmbdijOM78srl5DI+jjbYcLyRg5AVv
+
+    print(os.urandom(30).encode('hex'))
+    # 6573ba5016f740b04a0af7b59fe47045c0b9d4f94b1f613eaa11d276bc5e
+
+Then you can use:
+
+.. code-block:: shell
+
+    $ http GET https://kinto.dev.mozaws.net/v1/ \
+        --auth "token:6573ba5016f740b04a0af7b59fe47045c0b9d4f94b1f613eaa11d276bc5e"
+
+or:
+
+.. code-block:: shell
+
+    $ http GET https://kinto.dev.mozaws.net/v1/ \
+        --auth "token:nkz6+hN9KVtXX6bZ9+RPof0NgF9hmm5gFhKpWbiM"
+
+
+How Kinto knows it is a valid Basic Auth token?
+-----------------------------------------------
+
+For each token, Kinto will calculate a unique user ID which is
+related to your Kinto instance ``user_hmac_secret`` configuration.
```

Comment:
this is redondant with http://kinto.readthedocs.org/en/latest/api/1.x/cliquet/authentication.html#basic-auth